### ISSUE TITLE
CircleCI workflow `releases set-commits` fails with object not found

### ISSUE THREAD

**Issue by toote** [getsentry/sentry-cli#1973]
### Environment

Running in CircleCI image cimg/python:3.8 using latest CLI (verified with 2.28.6 and 2.30.0):

### Steps to Reproduce

* Create a the steps for releases as described in the official documentation to do so from CircleCI https://docs.sentry.io/product/releases/setup/release-automation/circleci/
* Run the CircleCI workflow

### Expected Result

Release is created as normal

### Actual Result

When running `sentry-cli releases set-commits $VERSION --auto --ignore-missing` the following error occurs:

```
error: object not found - cannot read header for (<SHA that is not the commit ID>); class=Odb (9); code=NotFound (-3)
```

The job had been working without issues and no changes 10 days ago.

### Logs

[sentry.log](https://github.com/getsentry/sentry-cli/files/14529334/sentry.log)

**Comment by alimony** [comment]
I wonder if something changed in Sentry, because I started seeing errors like these in workflows just a few hours ago:

```
Sentry CLI Plugin: Command failed: /home/runner/work/project/project/frontend/node_modules/@sentry/cli/sentry-cli releases set-commits project@1.0.5 --commit org/project@17d4cb0f1d7d089ad123e31007f3abc188bc123
error: API request failed
  caused by: sentry reported an error: You do not have permission to perform this action. (http status: 403)
```

**Comment by szokeasaurusrex** [comment]
@toote, does downgrading to an older Sentry CLI version solve your problem? If not, I would strongly suspect that the problem is originating in Sentry, rather than the CLI, as @alimony suggested.

**Comment by toote** [comment]
> @toote, does downgrading to an older Sentry CLI version solve your problem?

Nope, it happens both with 2.30.0 and 2.28.6, the latter which was working a week ago or so. 

> If not, I would strongly suspect that the problem is originating in Sentry, rather than the CLI, as @alimony suggested.

That was my guess as well, but sentry support asked me to open the report here in the case I created with them 🤷‍♂️ (115620 if anyone is interested)

**Comment by szokeasaurusrex** [comment]
@toote Thank you for the clarification. 

> That was my guess as well, but sentry support asked me to open the report here in the case I created with them 🤷‍♂️ (115620 if anyone is interested)

If you are certain that the command was working a week ago on version 2.28.6 but the command has stopped working in the same version now, I think the only plausible explanations are either a change in the inputs (configuration or input files) to the CI script, or a change in Sentry. Since you claim that you made no changes since the last time the script worked, the only remaining plausible explanation is a change in Sentry. Therefore, I am going to transfer this issue to the Sentry repository, so you can get help from the team responsible for maintaining the Sentry codebase.
